### ISSUE TITLE
Circle CI tests broken due to `test_version` test

### ISSUE THREAD

**Issue by sloosvel** [ESMValGroup/ESMValCore#1619]
**Describe the bug**
A test to check the version number for the tool was added in #1578 . This test checks that a proper version number has been generated by using a regex that expects the number to be defined as `X.X.Xdev1+etc` . However, in the CircleCI tests, the version number that gets generated looks like this: 

`0.1.dev1+g23eb7c8'`

This breaks the test because the version number only contains two digits separated by dots intead of three.

I guess the regex patter should be updated to also handle this format?

**Comment by bouweandela** [comment]
I don't think so, that looks like a wrong version number. It should start with 2.5.1. Do you have an example of a failing run? The [nightly](https://app.circleci.com/pipelines/github/ESMValGroup/ESMValTool/7197/workflows/84377f89-3688-4118-a8dd-d173777ea9c8) test run worked just fine.

**Comment by bouweandela** [comment]
You can recognize CircleCI tests from the URL, it links to circleci.com (instead of github.com).

**Comment by sloosvel** [comment]
>Those are GitHub Action test runs, not CircleCI.

Sorry, my bad!

>You can recognize CircleCI tests from the URL, it links to circleci.com (instead of github.com).

Thanks!

So I guess it's not a big deal if those tests are failing?

**Comment by bouweandela** [comment]
They need to be fixed of course, as they test e.g. that our code is compatible with all Python versions that we claim to support. Probably it is as easy as setting `fetch-depth: 0` in the [checkout action](https://github.com/actions/checkout#usage).

**Comment by sloosvel** [comment]
I think I don't have permissions to do so.

**Comment by bouweandela** [comment]
You can just make a pull request, the files are in the.github/workflows directory. Some documentation is available here: https://docs.esmvaltool.org/projects/esmvalcore/en/latest/contributing.html#automated-testing

**Comment by sloosvel** [comment]
Is it enough to just add it to the `run-tests` files? Not all workflows are failing.

**Comment by bouweandela** [comment]
It would be best to update all of them, to avoid surprises later.

**Comment by valeriupredoi** [comment]
Sorry guys, am on holidays - just saw this - it'd be best to pick up the code version using the Guthub API - that's the most secure and error-free way, but it escapes me atm what was the API invocation to get it, I know the PyPi package build test checks it (so it actually does the release if the version changed), so look there 🍺 Greetings from Transylvania 🧛
